exporter/datadogexporter/factory.go

```diff
@@ -39,6 +40,12 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/resourcetotelemetry"
 )
 
+var logsAgentExporterFeatureGate = featuregate.GlobalRegistry().MustRegister(
+	"exporter.datadogexporter.logsagentexporter",
+	featuregate.StageAlpha,
+	featuregate.WithRegisterDescription("When enabled, datadogexporter uses the Datadog agent logs pipeline for exporting logs."),
```

Comment:
Can we also add [`WithRegisterFromVersion`](https://pkg.go.dev/go.opentelemetry.io/collector/featuregate#WithRegisterFromVersion)?